cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -68,115 +54,120 @@ public int getDiffSize() {
-     * We want to skip "root" logger as well to avoid duplicated key in alterConfigOps collection.
-     * @param alterConfigOps collection of AlterConfigOp
-     * @param pathValueWithoutSlash name of "removed" logger
-     * @param entry entry to be removed (set to ERROR)
-     */
-    private static void removeProperty(Collection<AlterConfigOp> alterConfigOps, String pathValueWithoutSlash, ConfigEntry entry) {
-        if (!pathValueWithoutSlash.contains("log4j.appender") && !pathValueWithoutSlash.equals("root") && !"ERROR".equals(entry.value())) {
-            alterConfigOps.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, "ERROR"), AlterConfigOp.OpType.SET));
-            log.trace("{} not set in desired, setting to ERROR", entry.name());
-        }
+        return updatedCE;
     }
 
     /**
      * @return whether the current config and the desired config are identical (thus, no update is necessary).
      */
     @Override
     public boolean isEmpty() {
-        return  diff.size() == 0;
+        return diff.size() == 0;
     }
 
-    private static boolean isValidLoggerLevel(String level) {
-        return VALID_LOGGER_LEVELS.contains(level);
+    /**
+     * This internal class calculates the logging level of an arbitrary category based on the logging configuration.
+     *
+     * It takes Log4j properties configuration in the form of a map of key:value pairs,
+     * where key is the category name, and the value is whatever comes to the right of '=' sign in log4j.properties,
+     * which is either a logging level, or a logging level followed by a comma, and followed by the appender name.
+     */
+    static class LoggingLevelResolver {
+
+        private final Map<String, String> config;
+
+        LoggingLevelResolver(Map<String, String> loggingConfig) {
+            this.config = loggingConfig;
+        }
+
+        /**
+         * The method that returns the logging level of the category
+         * based on logging configuration, taking inheritance into account.
+         *
+         * For example, if looking for a logging level for 'io.strimzi.kafka.oauth.server.OAuthKafkaPrincipalBuilder',
+         * the following configuration lookups are performed until one is found:
+         * <ul>
+         *     <li>io.strimzi.kafka.oauth.server.OAuthKafkaPrincipalBuilder</li>
+         *     <li>io.strimzi.kafka.oauth.server</li>
+         *     <li>io.strimzi.kafka.oauth</li>
+         *     <li>io.strimzi.kafka</li>
+         *     <li>io.strimzi</li>
+         *     <li>io</li>
+         *     <li>root</li>
+         * </ul>
+         *
+         * @param name The logging category name
+         * @return The logging level
+         */
+        LoggingLevel resolveLevel(String name) {
+            String level = config.get(name);
+            if (level != null) {
+                return LoggingLevel.valueOf(level.split(",")[0]);
+            }
+
+            int e = name.length();
```

Comment:
Wouldn't it better to use this name or am I missing some meaning of `e`? )
```suggestion
            int loggingCategoryNameLength = name.length();
```

Reply:
Not really. Long variables make code harder to read so they should be used judiciously. Here 'e' really stands for 'end index', but seems to me, that should be obvious.